internal/provider/aws/provider_test.go

```diff
@@ -0,0 +1,4154 @@
+										GroupName: aws.String("sg-02"),
+									},
+								},
+							},
+							{
+								NetworkInterfaceId: aws.String(eni01ID),
+								Attachment: &types.NetworkInterfaceAttachment{
+									AttachmentId: aws.String("eni-attach-1" + testID),
+									InstanceId:   aws.String("i-1-" + testID),
+								},
+								Groups: []types.GroupIdentifier{
+									{
+										GroupId:   aws.String(groupID),
+										GroupName: aws.String(sgName),
+									},
+									{
+										GroupId:   aws.String("sg-01"),
+										GroupName: aws.String("sg-01"),
+									},
+									{
+										GroupId:   aws.String("sg-02"),
+										GroupName: aws.String("sg-02"),
+									},
+								},
+							},
+						},
+					}, nil
+				})
+
+			mockec2Client.EXPECT().
+				ModifyNetworkInterfaceAttribute(ctx, gomock.AssignableToTypeOf(&ec2.ModifyNetworkInterfaceAttributeInput{})).
+				DoAndReturn(func(_ context.Context, input *ec2.ModifyNetworkInterfaceAttributeInput, _ ...func(*ec2.Options)) (*ec2.ModifyNetworkInterfaceAttributeOutput, error) {
+					switch aws.ToString(input.NetworkInterfaceId) {
+					case eniID:
+						Expect(input.Groups).To(ConsistOf("sg-01", "sg-02"))
+						return &ec2.ModifyNetworkInterfaceAttributeOutput{}, nil
+					case eni01ID:
+						Expect(input.Groups).To(ConsistOf("sg-01", "sg-02"))
+						return &ec2.ModifyNetworkInterfaceAttributeOutput{}, fmt.Errorf("modify network interface attribute error")
+					default:
+						return nil, fmt.Errorf("unexpected interface ID: %s", aws.ToString(input.NetworkInterfaceId))
+					}
+				}).Times(2)
+
+			err := p.ReconcileFirewallRulesDeletion(ctx, log, nodeName, "")
+			Expect(err).To(MatchError(&provider.Error{
+				Code: "InternalError",
+				Msg:  "failed to modify network interface attribute: modify network interface attribute error",
+			}))
+		})
+
+		It("should return an error when an AWS API call (DeleteSecurityGroup) fails", func() {
+			mockec2Client.EXPECT().
+				DescribeSecurityGroups(ctx, gomock.AssignableToTypeOf(&ec2.DescribeSecurityGroupsInput{})).
+				DoAndReturn(func(_ context.Context, input *ec2.DescribeSecurityGroupsInput, _ ...func(*ec2.Options)) (*ec2.DescribeSecurityGroupsOutput, error) {
+					Expect(input.Filters).To(ConsistOf(filters))
+					return &ec2.DescribeSecurityGroupsOutput{
+						SecurityGroups: []types.SecurityGroup{
+							{
+								GroupId:   aws.String(groupID),
+								GroupName: aws.String(sgName),
+							},
+						},
+					}, nil
+				})
+			mockec2Client.EXPECT().
+				DescribeNetworkInterfaces(ctx, gomock.AssignableToTypeOf(&ec2.DescribeNetworkInterfacesInput{})).
+				DoAndReturn(func(_ context.Context, input *ec2.DescribeNetworkInterfacesInput, _ ...func(*ec2.Options)) (*ec2.DescribeNetworkInterfacesOutput, error) {
+					Expect(input.Filters).To(HaveLen(1))
+					Expect(aws.ToString(input.Filters[0].Name)).To(Equal("group-id"))
+					Expect(input.Filters[0].Values).To(ConsistOf(groupID))
```

Comment:
Idem, you forgot to fix https://github.com/quortex/kubestatic/pull/186#discussion_r2060264221 here